### PR TITLE
Added option to use APCu instead of APC

### PR DESCRIPTION
**Pull request by SpaceCafe**
screenshot apcu

### CODE REVIEW

File: protected/humhub/tests/codeception/fixtures/data/setting.php [hunk 1]
```diff
@@ -17,6 +17,7 @@
     array('name' => 'maxFileSize', 'value' => '1048576', 'module_id' => 'file'),
     array('name' => 'forbiddenExtensions', 'value' => 'exe', 'module_id' => 'file'),
     array('name' => 'cache.class', 'value' => 'CFileCache', 'module_id' => 'base'),
+    array('name' => 'cache.useApcu', 'value' => '0', 'module_id' => 'base'),
```
Comment (Felli) [inline protected/humhub/tests/codeception/fixtures/data/setting.php]: Shouldn't this also work? Just trying to understand why `0` would be the better choice.
`array('name' => 'cache.useApcu', 'value' => 'CFileCache', 'module_id' => 'base'),`
Or this should also work being Yii2.
`array('name' => 'cache.useApcu', 'value' => 'FileCache', 'module_id' => 'base'),`

File: protected/humhub/libs/DynamicConfig.php [hunk 2]
```diff
@@ -109,9 +109,11 @@ public static function rewrite()
 
         // Add Caching
         $cacheClass = Yii::$app->settings->get('cache.class');
+        $cacheUseApcu = Yii::$app->settings->get('cache.useApcu');
         if (in_array($cacheClass, ['yii\caching\DummyCache', 'yii\caching\ApcCache', 'yii\caching\FileCache'])) {
             $config['components']['cache'] = [
                 'class' => $cacheClass,
+                'useApcu' => $cacheUseApcu,
```
Comment (cebe) [inline protected/humhub/libs/DynamicConfig.php]: this will break FileCache and DummyCache as they do not have this property.

Comment (Felli) [inline protected/humhub/libs/DynamicConfig.php]: Hmm... Would `'class' => $cacheUseApcu,` work for this?

Comment (cebe) [inline protected/humhub/libs/DynamicConfig.php]: no, that does not make sense, the if needs to have different branches, one for Apc and one for DummyCache and FileCache.

```php
         if (in_array($cacheClass, ['yii\caching\DummyCache', 'yii\caching\FileCache'])) {
             $config['components']['cache'] = [
                 'class' => $cacheClass,
                 'keyPrefix' => Yii::$app->id
             ];
         } elseif ($cacheClass === 'yii\caching\ApcCache') {
             $config['components']['cache'] = [
                 'class' => $cacheClass,
                 'useApcu' => $cacheUseApcu,
                 'keyPrefix' => Yii::$app->id
             ];
         }
```

**Comment by luke-** [comment]
Thanks! We decided to automatically switch to APCu when available.

**Pull request closed without merging**